Calibration decoupling

amrsoll opened the pull request:
Make the Corner and Lens calibration steps independent !

amrsoll committed:
Decoupling : Stage 1

- Detect markers on the raw picture
- Undistort the picture after the detection
- Add some dictionnary tools
- Add documentation
- Can now predict a pixel position after undistort
- Possibility to up/downscale image
- TODO : Could take a different picture resolution.

TODO : Calibraton tool should save:
- the raw position of the arrow tip on the image
- Convert that position through lens distortion for backwards compatibility

amrsoll committed:
Decoupling: Stage 2

- Corner calibration saves in the correct locations
- Optimisation : Turn the image grey before lens correction
- yaml dump should use the default (more readable) dumper

amrsoll committed:
Decoupling : Stage 3

- Refactoring
  - definitions file to store constants separately
  - corner and lens calibration separated into 2 files
  - lid handler : serve picture gets shorter
- Clean up some file saving feature

ghost commented:
## DeepCode's analysis on [#46d89b](https://www.deepcode.ai/app/gh/mrbeam/MrBeamPlugin/db45cff654eb10277d910331025759bbaa7f8307/mrbeam/MrBeamPlugin/46d89bec783c11b0e556d253e4d18e5981a09f11/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=2&) found:

- :information_source: **2** minor issues. :point_down:
- :heavy_check_mark: **2** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">No need to call keys, directly check with the in operator. Occurrences: <ul><li><a href="https://github.com/mrbeam/MrBeamPlugin/blob/46d89bec783c11b0e556d253e4d18e5981a09f11/octoprint_mrbeam/camera/corners.py#L144">corners.py:144</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/mrbeam/MrBeamPlugin/db45cff654eb10277d910331025759bbaa7f8307/mrbeam/MrBeamPlugin/46d89bec783c11b0e556d253e4d18e5981a09f11/pr/_/%2Foctoprint_mrbeam%2Fcamera%2Fcorners.py/python%2Fdc%2Freplace~keys~list~compare/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Using the loop variable _k from some outer or global scope in this for of loop may prevent optimisations by the JavaScript engine. Consider declaring the loop variable inside the head of this for of loop. Occurrences: <ul><li><a href="https://github.com/mrbeam/MrBeamPlugin/blob/46d89bec783c11b0e556d253e4d18e5981a09f11/octoprint_mrbeam/static/js/calibration/camera_calibration.js#L884">camera_calibration.js:884</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/mrbeam/MrBeamPlugin/db45cff654eb10277d910331025759bbaa7f8307/mrbeam/MrBeamPlugin/46d89bec783c11b0e556d253e4d18e5981a09f11/pr/_/%2Foctoprint_mrbeam%2Fstatic%2Fjs%2Fcalibration%2Fcamera_calibration.js/javascript%2Fdc%2FNonLocalLoopVar/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/mrbeam/MrBeamPlugin/db45cff654eb10277d910331025759bbaa7f8307/mrbeam/MrBeamPlugin/46d89bec783c11b0e556d253e4d18e5981a09f11/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=2&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=mrbeam)

amrsoll committed:
Decoupling : Final stage

- All the constants imported from the definitions file
- MrbPicWorker defined in worker.py
- Removed typing dependency (unavailable on our machines)
- Moved lens undistorion functions from init and undistort.py to lens.py

amrsoll committed:
Change coords to x,y instead of i,j + dbg imgs.

amrsoll committed:
Add camera unit test with pytest.

amrsoll committed:
Move camera worker to separate module + pytest.ini

amrsoll committed:
Fix i j to x y coords.

amrsoll committed:
Rm test outputfile

amrsoll committed:
Ignore debug files from tests.

amrsoll committed:
Extra debug images and pytest debug.

amrsoll commented:
I don't really understand why the bot complains about the _k variable in js aber bon...

amrsoll commented:
Would be merged with the front-end part
